### PR TITLE
Adjustments for RGW as the default backingstore over 4.4 Azure

### PR DESCRIPTION
**Pull request by ebenahar**
Signed-off-by: Elad Ben Aharon <ebenahar@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 1]
```diff
@@ -130,6 +130,12 @@ def ocs_install_verification(
             post_upgrade_verification
         ) else 1
 
+    # With 4.4 OCS cluster deployed over Azure, RGW is the default backingstore
+    if float(
+        config.ENV_DATA['ocs_version']
+    ) in [4.4, 4.5] and config.ENV_DATA.get('platform') == constants.AZURE_PLATFORM:
```
Comment (mbukatov) [inline ocs_ci/ocs/resources/storage_cluster.py:136]: There should be no RGW on Azure with OCS 4.5.

Comment (ebenahar) [reply ocs_ci/ocs/resources/storage_cluster.py:136]: but after the upgrade to 4.5, there will

**Comment by ebenahar** [comment]
@mbukatov not sure this is actually a workaround, as RGW will continue to be the default backingstore in 4.4

File: ocs_ci/ocs/resources/storage_cluster.py (hunk 1 above)
Comment (petr-balogh) [reply ocs_ci/ocs/resources/storage_cluster.py:136]: then we need to update this check to be valid only if cluster is after upgrade right?

Comment (ebenahar) [reply ocs_ci/ocs/resources/storage_cluster.py:136]: @mbukatov @petr-balogh right, will fix

File: ocs_ci/ocs/resources/mcg.py [hunk 2]
```diff
@@ -143,10 +143,24 @@ def __init__(self, *args, **kwargs):
             pods = pod.get_pods_having_label(label=constants.RGW_APP_LABEL, namespace=self.namespace)
```
Comment (petr-balogh) [inline ocs_ci/ocs/resources/mcg.py:143]: Can you also please change the line above from:
logger.info('Checking whether RGW pod is not present on AWS platform')

to something like:
logger.info('Checking whether RGW pod is not present.')
or mention all possible cases like not present on cloud providers, external mode?  If the storagecluster_independent_check means it's external mode :)

Comment (ebenahar) [reply ocs_ci/ocs/resources/mcg.py]: done

**Comment by mbukatov** [comment]
> @mbukatov not sure this is actually a workaround, as RGW will continue to be the default backingstore in 4.4

It is change in ocs-ci code to overcome known product BZ 1825994 *RGW pods should not be started in cloud deployments like Azure, GCP* - so it seems like [a workaround](https://ocs-ci.readthedocs.io/en/latest/docs/workarounds.html) to me. But we don't plan to remove it as the checks are version specific, so we don't have to create issue to remove it.

File: ocs_ci/ocs/resources/storage_cluster.py (hunk 1 above)
Comment (ebenahar) [reply ocs_ci/ocs/resources/storage_cluster.py]: this is already being done

> > @mbukatov not sure this is actually a workaround, as RGW will continue to be the default backingstore in 4.4
> 
> It is change in ocs-ci code to overcome known product BZ 1825994 _RGW pods should not be started in cloud deployments like Azure, GCP_ - so it seems like [a workaround](https://ocs-ci.readthedocs.io/en/latest/docs/workarounds.html) to me. But we don't plan to remove it as the checks are version specific, so we don't have to create issue to remove it.

Exactly

**Review by petr-balogh — APPROVED** [review]

**Comment by ebenahar** [comment]
> > @mbukatov not sure this is actually a workaround, as RGW will continue to be the default backingstore in 4.4
> 
> It is change in ocs-ci code to overcome known product BZ 1825994 _RGW pods should not be started in cloud deployments like Azure, GCP_ - so it seems like [a workaround](https://ocs-ci.readthedocs.io/en/latest/docs/workarounds.html) to me. But we don't plan to remove it as the checks are version specific, so we don't have to create issue to remove it.

Exactly

**Comment by petr-balogh** [comment]
Running verification here: https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-azure-ipi-1az-rhcos-3m-3w-upgrade-ocs-auto/4/console

**Comment by petr-balogh** [comment]
@ebenahar so, the deployment and upgrade has passed so I think we can mark as verified.

```
$ oc get pod -n openshift-storage
NAME                                                              READY   STATUS      RESTARTS   AGE
csi-cephfsplugin-provisioner-777b6c59df-6lstm                     5/5     Running     0          163m
csi-cephfsplugin-provisioner-777b6c59df-wvmn5                     5/5     Running     0          163m
csi-cephfsplugin-s27m2                                            3/3     Running     0          163m
csi-cephfsplugin-vfflh                                            3/3     Running     0          162m
csi-cephfsplugin-w6dc2                                            3/3     Running     0          162m
csi-rbdplugin-5hnf5                                               3/3     Running     0          163m
csi-rbdplugin-provisioner-6bcd8df965-tgjfm                        5/5     Running     0          163m
csi-rbdplugin-provisioner-6bcd8df965-tzjb2                        5/5     Running     0          163m
csi-rbdplugin-tglj2                                               3/3     Running     0          162m
csi-rbdplugin-zbvp4                                               3/3     Running     0          162m
noobaa-core-0                                                     1/1     Running     0          163m
noobaa-db-0                                                       1/1     Running     0          163m
noobaa-endpoint-b44476596-8t8tq                                   1/1     Running     0          163m
noobaa-endpoint-b44476596-xshlf                                   1/1     Running     0          163m
noobaa-operator-69766f8f76-hgbjz                                  1/1     Running     0          164m
ocs-operator-667584f75c-mqg2x                                     1/1     Running     0          164m
rook-ceph-crashcollector-10b37608eea92e5f8a31625038441ca2-6ps52   1/1     Running     0          3h39m
rook-ceph-crashcollector-20452bc58c059762fef7824abcede560-24bsd   1/1     Running     0          3h39m
rook-ceph-crashcollector-91adb7fb28b37a12da14673f3b51378c-rft9f   1/1     Running     0          162m
rook-ceph-crashcollector-a0f796e95aa3635f89fe03afad020ee1-cfxth   1/1     Running     0          3h35m
rook-ceph-crashcollector-aba782021505274efa99f4b2504f4545-htgrn   1/1     Running     0          161m
rook-ceph-crashcollector-f33f183eb6ed3c0450c9297eee4c8a86-6sgsh   1/1     Running     0          162m
rook-ceph-drain-canary-jnk-zi1c33-ua-522wd-worker-eastus1-xt6zl   1/1     Running     0          158m
rook-ceph-drain-canary-jnk-zi1c33-ua-522wd-worker-eastus2-zksbv   1/1     Running     0          157m
rook-ceph-drain-canary-jnk-zi1c33-ua-522wd-worker-eastus3-blmwv   1/1     Running     0          155m
rook-ceph-mds-ocs-storagecluster-cephfilesystem-a-db5c4bdcw5mtx   1/1     Running     0          162m
rook-ceph-mds-ocs-storagecluster-cephfilesystem-b-5b75d7c5xnf66   1/1     Running     0          161m
rook-ceph-mgr-a-858999846-mv7rl                                   1/1     Running     0          159m
rook-ceph-mon-a-9f87846db-qhl7m                                   1/1     Running     0          161m
rook-ceph-mon-b-6fb8466444-bp82h                                  1/1     Running     0          159m
rook-ceph-mon-c-f7bc589c4-b7p8l                                   1/1     Running     0          162m
rook-ceph-operator-9b4c4684d-qjrg9                                1/1     Running     0          164m
rook-ceph-osd-0-849b9d8d97-5fmnc                                  1/1     Running     0          158m
rook-ceph-osd-1-66dd685976-rlzvq                                  1/1     Running     0          155m
rook-ceph-osd-2-69f7878b46-wpfrk                                  1/1     Running     0          157m
rook-ceph-osd-3-78f6c5ddf8-bls6q                                  1/1     Running     0          124m
rook-ceph-osd-4-66c5bc69cd-gps69                                  1/1     Running     0          124m
rook-ceph-osd-5-7bb6dd6787-lmht2                                  1/1     Running     0          124m
rook-ceph-osd-prepare-ocs-deviceset-0-0-pgm79-msltt               0/1     Completed   0          3h40m
rook-ceph-osd-prepare-ocs-deviceset-0-data-1-rc6f5-87gbx          0/1     Completed   0          125m
rook-ceph-osd-prepare-ocs-deviceset-1-0-rt9p7-rt9fh               0/1     Completed   0          3h40m
rook-ceph-osd-prepare-ocs-deviceset-1-data-1-swxp6-lsr4j          0/1     Completed   0          125m
rook-ceph-osd-prepare-ocs-deviceset-2-0-pgj6x-sxskn               0/1     Completed   0          3h40m
rook-ceph-osd-prepare-ocs-deviceset-2-data-1-2qt4z-b28d2          0/1     Completed   0          125m
rook-ceph-rgw-ocs-storagecluster-cephobjectstore-a-6f46597rh7vk   1/1     Running     0          162m
rook-ceph-tools-5f9d64c99b-wgbp9                                  1/1     Running     0          3h35m

$ oc get csv -n openshift-storage
NAME                                           DISPLAY                       VERSION                 REPLACES              PHASE
elasticsearch-operator.4.5.0-202008100413.p0   Elasticsearch Operator        4.5.0-202008100413.p0                         Succeeded
ocs-operator.v4.5.0-70.ci                      OpenShift Container Storage   4.5.0-70.ci             ocs-operator.v4.4.2   Succeeded```

I see there is RGW pod: `rook-ceph-rgw-ocs-storagecluster-cephobjectstore-a-6f46597rh7vk` after upgrade.

**Review by clacroix12 — APPROVED** [review]

**Pull request merged**